### DISCUSSION TITLE
On the issue about receiving NaN values in log

### DISCUSSION THREAD

**Discussion by seuflyer** [bitcraze/discussions#1372] [Q&A]
I added my own group to the firmware library and added a variable 's' to this group：

```c
float myLogValueFunction1(uint32_t timestamp, void* data) {
        // Return the log value
  return s;
}

logByFunction_t myLogger1 = {.aquireFloat = myLogValueFunction1, .data = 0};

LOG_GROUP_START(myGroup)
LOG_ADD_BY_FUNCTION(LOG_FLOAT, myLog1, &myLogger1)
LOG_GROUP_STOP(myGroup)
```

Then I used Python code to subscribe to it, with a cycle of 10ms，
But the result obtained is a null value. It is interesting that if a constant is sent, such as 10, it can be successfully received
<img width="416" alt="图片1" src="https://github.com/bitcraze/discussions/assets/171215615/39675fa8-b131-4c3b-b520-3037c4363bf3">

**Comment by gemenerik** [comment]
Can you give me some more information about what `s` is? Do you initialize it? Is it a float like you define in `LOG_ADD_BY_FUNCTION(LOG_FLOAT, myLog1, &myLogger1)`?

This as an app works just fine for me:
```c
#include <string.h>
#include <stdint.h>
#include <stdbool.h>

#include "app.h"

#include "FreeRTOS.h"
#include "task.h"

#include "debug.h"

#include "log.h"

float s;

float myLogValueFunction(uint32_t timestamp, void* data) {
    // Return the log value
    return s;
}

void appMain()
{
  s = 13.2;

  while(1) {
    vTaskDelay(M2T(1000));
    s += 0.1f;
  }
}

logByFunction_t myLogger = {.aquireFloat = myLogValueFunction, .data = 0};

LOG_GROUP_START(myGroup)
LOG_ADD_BY_FUNCTION(LOG_FLOAT, myLog, &myLogger)
LOG_GROUP_STOP(myGroup)
```
